src/dioptra/restapi/v1/workflows/service.py

```diff
@@ -65,3 +67,63 @@ def get(self, job_id: int, file_type: FileTypes, **kwargs) -> IO[bytes]:
             file_type=file_type,
             logger=log,
         )
+
+
+class SignatureAnalysisService(object):
+    """The service methods for performing signature analysis on a file."""
+
+    def post(self, fileContents: str, **kwargs) -> dict[str, List[dict[str, Any]]]:
```

Comment:
Stick with snakecase, and we can use `list` instead of `List` in the type annotation:

```suggestion
    def post(self, file_contents: str, **kwargs) -> dict[str, list[dict[str, Any]]]:
```

The `file_contents` argument name might also need to be changed after addressing the suggestions in workflows/schema.py.